core/src/test/java/io/parsingdata/metal/ByteLengthTest.java

```diff
@@ -53,8 +54,8 @@
     // but Len will become useful when Let is implemented
     private static final Token STRING = seq(
         def("length", 1),
-        def("text1", ref("length")),
-        def("text2", len(ref("text1"))));
+        def("text1", last(ref("length"))),
+        def("text2", last(len(ref("text1")))));
```

Comment:
(You mentioned this offline, and I must it does create some noise...)

Reply:
For len it will dissappear I guess. For ref, well... Java does support return type overloading on the bytecode level :smiling_imp: